converters/toZigbee.js

```diff
@@ -854,19 +883,26 @@ const converters = {
                 // Simulate 'On' with transition via 'MoveToLevelWithOnOff', or restore the level from before
                 // it was clobbered by a previous transition to off; otherwise just use 'On'.
                 // TODO: same problem as above.
-                // TODO: if transition is not specified, should use device default (OnTransitionTime), not 0.
                 if (transition.specified || globalStore.getValue(entity, 'turnedOffWithTransition') === true) {
-                    const current = utils.getObjectProperty(meta.state, 'brightness', 254);
-                    brightness = globalStore.getValue(entity, 'brightness', current);
-                    try {
-                        const attributeRead = await entity.read('genLevelCtrl', ['onLevel']);
-                        // TODO: for groups, `read` does not wait for responses. If it did, we could still issue a single
-                        //  command if all values of `OnLevel` are equal, or split into one command per device if not.
-                        if (attributeRead !== undefined && attributeRead['onLevel'] != 255) {
-                            brightness = attributeRead['onLevel'];
+                    if (!meta.state.hasOwnProperty('level_config')) {
+                        await converters.level_config.convertGet(entity, 'level_config', meta);
```

Comment:
If I understand correctly, this will read the `level_config` for all bulbs when turned on; are we sure that all bulbs support this?

Reply:
This code only executes when we can't use On because CurrentLevel got clobbered or we need a custom transition duration. MoveToLevelWithOnOff does not default to OnTransitionTime, so we have to read it manually.

>are we sure that all bulbs support this?

level_config.convertGet ignores any unsupported attributes, so we'll just default to on_transition_time = "disabled", on_level = "previous", and on_off_transition_time = 0 if any of them are unsupported.

Reply:
This code is somewhat questionable though, because
1. if you manually set part of level_config (e.g. execute_if_off), this code will never read the rest of it and assume other attributes are unsupported;
2. the read only takes effect the *second* time its results are needed, I'm guessing because `meta.state` we already have is immutable and not affected by the read
Not sure what to do about that.

Reply:
Fixed that, I think. Also disabled reading level_config for groups, since it's not merged from members' states anyway.

Reply:
> MoveToLevelWithOnOff does not default to OnTransitionTime, so we have to read it manually. If level_config.execute_if_off is enabled we can probably use MoveToLevel + On instead, though in case of custom transition OnLevel still needs to be read. 

What about not caring about the default transition time in the bulb? You can set a default transition time in the zigbee2mqtt config like this; https://www.zigbee2mqtt.io/devices/046677476816.html#options

Reply:
I suppose that's an option too. In that case `transition.specified` is simply always true. I suspect it'd interact poorly with binding though - not only do you have to sync the attribute and the config option, turning the bulb off via z2m then on via a bound remote will consistently set it to minimum brightness because all z2m off commands will have a transition.

Reply:
Ok, I guess it's not really necessary.